### ISSUE TITLE
Mangled generated code mixing ExportAllDeclaration and external source

### ISSUE THREAD

**Issue by usergenic** [rollup/rollup#2489]
- Rollup Version: 0.64.0
- Operating System (or Browser):
- Node Version: 10.9.0

### How Do We Reproduce?

Here's a fairly simple reproduction, call it `repro.js`
```js
var { rollup } = require('rollup');

const files = {
  'file:///a.js': `import * as $b from "file:///b.js"; export { $b };`,
  'file:///b.js': `export * from 'file:///c.js';`,
  'file:///c.js': `export const c = '🤡';`
};

rollup({
  input: 'file:///a.js',
  external: ['file:///c.js'],
  output: { format: 'esm' },
  plugins: [{ name: "somePlugin", resolveId: (id) => id, load: (id) => files[id] }]
}).then((r) => r.generate({ format: 'es' }).then((r) => console.log(r.code)));
```

Run the repro:
```bash
npm install rollup@0.64.0
node repro.js
```

### Expected Behavior

I'd expect something along the lines of the following, but would be happy with anything that achieves 

```js
import * as c from 'file:///c.js';
var $b = {
  c: c
};
export { $b };
```

### Actual Behavior

Generates the following mess.
```js
import 'file:///c.js';



var b = /*#__PURE__*/Object.freeze({
  *file:///c.js: *
});

export { b as $b };
```

**Comment by lukastaegert** [comment]
This is indeed a tricky one. Your expected result is wrong IMO since if you do

```js
import * as c from 'file:///c.js';
```

with `file:///c.js` as given in your example, then the namespace variable `c` would have the form `{c: 🤡}`. Thus in your example, `$b$` should have this form as well while in your proposed solution, it would have the form `{c: {c: 🤡}}`.

A valid solution would be

```js
import * as c from 'file:///c.js';
export { c as $b };
```

However, this is no longer possible if `file:///b.js` adds other imports, e.g.

```js
// b.js
export * from 'file:///c.js';
export const dontNoWhatToDoWithThis = 4;
```

My feeling is that we should try to use the first solution if possible but throw an error in the second case until someone figures out a nice solution that scales well.

**Comment by lukastaegert** [comment]
Interestingly enough, rollup 0.66 refuses to bundle this code with an internal error...

**Comment by lukastaegert** [comment]
Not really a fix but a better error message: #2499
